### PR TITLE
Added naive String setter

### PR DESCRIPTION
**Pull request by kleinjakob**
Added a naive string setter using the SimConnect py binding of the exported function according to SimConnect SDK.

Seems to work on ATC_ID...

Extensive testing on other Variables still needed.

### CODE REVIEW

**Comment by odwdinc** [comment]
Wow, Thanks for this!
Let me know when you are ready for review/pull.

File: SimConnect/SimConnect.py [hunk 1]
```diff
@@ -173,8 +173,14 @@ def request_data(self, _Request):
 		_Request.LastID = temp.value
 
 	def set_data(self, _Request):
-		pyarr = list([_Request.outData])
-		dataarray = (ctypes.c_double * len(pyarr))(*pyarr)
+		rtype = _Request.definitions[0][1].decode()
+		if 'String' in rtype or 'string' in rtype:
+			pyarr = bytearray(_Request.outData)
+			dataarray = (ctypes.c_char * len(pyarr))(*pyarr)
```
Comment (kleinjakob) [inline SimConnect/SimConnect.py]: Have to check for requests that are longer than the 'String32' datatype from the C SDK.

**Comment by kleinjakob** [comment]
Sorry for the bad quality... but it suffices to show what i've done.

From my simple REPL it seems setting Strings works. Sofar just the empty string of length 0 seems to need handling seperately.

rec

p.s. also controlling pushback via TOGGLE_PUSHBACK and KEY_TUG_HEADING from a command line is a lot of fun :-D

**Comment by odwdinc** [comment]
This is grate work! Thanks for sharing!

**Comment by kleinjakob** [comment]
Actually it is not necessarily the bytes of length 0, rather when I try to push a String thats "visually" empty e.g. `' '.encode('ascii')` (i.e a space 0x20) i get the Exception "SIMCONNECT_EXCEPTION_SIZE_MISMATCH" and a freeze in the simulator. otherwise even too long strings are gracefully handled and just cut off (ATC_ID can store up to 9 characters, everything beyond that is ignored, no exeption, no freeze or crash!

Still working on it will let you know once its ready to review.

**Review by kleinjakob — COMMENTED** [review]
longer than String32 is not a problem but "non printing chars" seem to cause problems...

**Comment by kleinjakob** [comment]
Tested quite a lot today, seems to work so far. I think its ready for review.

**Comment by odwdinc** [comment]
Amazing work, Thought I would try 
```
atcid = aq.find("ATC_ID")
print(atcid.get())
atcid.set("OD_🔥".encode())
print(atcid.get())
```
image

No Go but funny,
thanks agene!

**Comment by kleinjakob** [comment]
I noticed that too... either the font used or text-renderer in MSFS doesnt support UTF-8 or (color) emojis. I remember Microsoft Windows traditionally having some issues with font encodings... –especially the Wide-Char (16bit) approach around 2000–but thats not an issue here i think as it correctly interprets ASCII .

Another but (I hope indeed less likely!) option would be that its really just an 8-bit interface supporting some ASCII or superset thereof (LATIN-1 or something like that).

We will be able to confirm UTF-8 support or not when we use a known font which is known to have the codepoint under consideration...

I will keep it on my radar

**Pull request merged**